api/v1beta1/azurecluster_webhook.go

```diff
@@ -75,6 +75,13 @@ func (c *AzureCluster) ValidateUpdate(oldRaw runtime.Object) error {
 		)
 	}
 
+	if !reflect.DeepEqual(c.Spec.ControlPlaneEndpoint, old.Spec.ControlPlaneEndpoint) {
```

Comment:
we should allow setting each value if it is empty in this case to preserve existing behavior since the controller might patch it after the cluster is created

Reply:
I thought I did that, but it missed somehow.